gems/smithy/lib/smithy/views/client/shapes.rb

```diff
@@ -69,41 +69,42 @@ def build_shape(id, shape)
             name: Model::Shape.name(id).camelize,
             type: shape_type_from_type(shape['type']),
             traits: filter_traits(shape['traits']),
-            members: build_member_shapes(shape)
+            members: build_member_shapes(id, shape)
           )
         end
 
-        def build_member_shapes(shape)
+        def build_member_shapes(id, shape)
           case shape['type']
           when 'enum', 'intEnum', 'structure', 'union'
-            build_members(shape)
+            build_members(id, shape)
           when 'list'
-            build_list_member(shape)
+            build_list_member(id, shape)
           when 'map'
-            build_map_members(shape)
+            build_map_members(id, shape)
           else
             []
           end
         end
 
-        def build_members(shape)
-          shape['members'].map { |k, v| build_member_shape(k, v['target'], v['traits']) }
+        def build_members(id, shape)
+          shape['members'].map { |k, v| build_member_shape(id, k, v['target'], v['traits']) }
         end
 
-        def build_list_member(shape)
+        def build_list_member(id, shape)
           m_shape = shape['member']
-          [] << build_member_shape('member', m_shape['target'], m_shape['traits'])
+          [] << build_member_shape(id, 'member', m_shape['target'], m_shape['traits'])
         end
 
-        def build_map_members(shape)
+        def build_map_members(id, shape)
           %w[key value].map do |m_name|
             m_shape = shape[m_name]
-            build_member_shape(m_name, m_shape['target'], m_shape['traits'])
+            build_member_shape(id, m_name, m_shape['target'], m_shape['traits'])
           end
         end
 
-        def build_member_shape(name, id, traits)
+        def build_member_shape(parent_id, name, id, traits)
```

Comment:
This was a lot of plumbing. I think parent id should be id and the current id renamed to target?

Reply:
calling it `id` for the parent shapes id is a little confusing to me - theres technically three "ids" we're talking about for a member: 1) the parent shapes id 2) the member shape's id and 3) the target of the member shape's id.

I'd prefer to keep this explicit.

Reply:
Isn't member shapes id the same as name here? I thought members didn't have IDs, only the shapes with members and their targets

Reply:
the member shape id includes the name eg: `smithy.example.foo#ExampleShapeName$memberName` see: https://smithy.io/2.0/spec/model.html#shape-id).  I don't think our codegen would ever really use member ids though.

I could rename the `id` parameter here as `target_id` as well?

Reply:
I think target is preferred naming either way, it's what we've been using.